modules/sdk-coin-apt/src/lib/transaction/transaction.ts

```diff
@@ -258,14 +267,24 @@ export abstract class Transaction extends BaseTransaction {
   }
 
   loadInputsAndOutputs(): void {
+    const totalAmount = this._recipients?.reduce(
+      (accumulator, current) => accumulator.plus(current.amount),
+      new BigNumber('0')
+    );
     this._inputs = [
       {
         address: this.sender,
-        value: this.recipient.amount as string,
+        value: totalAmount?.toString() || (this.recipient.amount as string),
         coin: this._coinConfig.name,
       },
     ];
-    this._outputs = [
+    this._outputs = this._recipients?.map((recipient) => {
```

Comment:
remove null checks from all places because `_recipients` will be initialized with empty list in the constructor.
```suggestion
    this._outputs = this._recipients.map((recipient) => {
```